### PR TITLE
Daemonset for Nvidia/CUDA installation

### PR DESCRIPTION
**Pull request by prasunanand**

### CODE REVIEW

**Comment by costrouc** [comment]
The daemonset looks right to me. Could we move this to a separate file in that directory? Also can we only create the resource if a user specifies a node group with gpus. You can control if a resource is created with `count = 0 or 1`. If `length(concat([for node_group in var.merged_nodegroups: nodegroup.guest_accelerators])) == 0 ? 0, 1` I think would be a working expression for count.

File: modules/gcp/kubernetes/nvidia_installer.tf [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+            resources = {
+              requests = {
+                cpu = 0.15
+              }
+            }
+            security_context = {
+              privileged = true
+            }
+            env = [
+              {
```
Comment (costrouc) [inline modules/gcp/kubernetes/nvidia_installer.tf:68]: I think that these must be written as

```hcl
env {
   name  = "NVIDIA_INSTALL_DIR_HOST"
   value = "/home/kubernetes/bin/nvidia"
}

...
```

I may be wrong.

File: modules/gcp/kubernetes/nvidia_installer.tf [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                name  = "VULKAN_ICD_DIR_CONTAINER"
+                value = "/etc/vulkan/icd.d"
+              },
+              {
+                name  = "ROOT_MOUNT_DIR"
+                value = "/root"
+              },
+            ]
+            volume_mounts = [
```
Comment (costrouc) [inline modules/gcp/kubernetes/nvidia_installer.tf:89]: Same goes with volume_mounts. I think they must be written as

```hcl
volume_mount {
  name  = "NVIDIA_INSTALL_DIR_HOST"
  value = "/home/kubernetes/bin/nvidia"
}
```

...

**Comment by prasunanand** [comment]
closed in favor of https://github.com/Quansight/qhub-terraform-modules/pull/16

**Pull request closed without merging**